editor/editor_node.cpp

```diff
@@ -2229,11 +2229,13 @@ void EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) {
 			}
 
 		} break;
-		case FILE_CLOSE_ALL_AND_QUIT:
-		case FILE_CLOSE_ALL_AND_RUN_PROJECT_MANAGER:
 		case FILE_CLOSE: {
+			_scene_tab_closed(editor_data.get_edited_scene());
```

Comment:
What if you have unsaved work?

Reply:
It will ask you if you want to save. It basically does what the code below do, but in a different place.

Reply:
Why do `FILE_CLOSE_ALL_AND_QUIT` and `FILE_CLOSE_ALL_AND_RUN_PROJECT_MANAGER` use a different logic then?

And what about the preview generation further down after the `[[fallthrough]]`?

Reply:
> Why do FILE_CLOSE_ALL_AND_QUIT and FILE_CLOSE_ALL_AND_RUN_PROJECT_MANAGER use a different logic then?

No idea tbh. This should be probably unified, see #40539

> And what about the preview generation further down after the [[fallthrough]]?

If it was generated without saving then it won't anymore. Closing current tab now works the same as middle-clicking it.